setup.py

```diff
@@ -8,23 +9,54 @@
 AMGX_DIR = os.environ.get('AMGX_DIR')
 AMGX_BUILD_DIR = os.environ.get('AMGX_BUILD_DIR')
 
+
+if sys.platform == "win32":
+    lib_name = 'amgxsh.dll'
+else:
+    lib_name = 'libamgxsh.so'
+
 if not AMGX_DIR:
     # look in PREFIX:
     PREFIX = sys.prefix
-    if os.path.isfile(os.path.join(PREFIX, 'lib/libamgxsh.so')):
+    if os.path.isfile(os.path.join(PREFIX, f'lib/{lib_name}')):
         AMGX_lib_dirs = [os.path.join(PREFIX, 'lib')]
         AMGX_include_dirs = [os.path.join(PREFIX, 'include')]
     else:
-        raise EnvironmentError('AMGX_DIR not set and libamgxsh.so not found')
+        raise EnvironmentError(f'AMGX_DIR not set and {lib_name} not found')
 else:
     if not AMGX_BUILD_DIR:
         AMGX_BUILD_DIR = os.path.join(AMGX_DIR, 'build')
-    AMGX_lib_dirs = [AMGX_BUILD_DIR]
+
+    for root, dirs, files in os.walk(AMGX_BUILD_DIR):
+        if lib_name in files:
+            lib_path = root
+            break
+    else:
+        raise RuntimeError(f'Cannot locate {lib_name} under "{AMGX_BUILD_DIR}".')
+
+    AMGX_lib_dirs = [lib_path]
     AMGX_include_dirs = [
         os.path.join(AMGX_DIR, 'base/include'),
         os.path.join(AMGX_DIR, 'core/include')
     ]
 
+lib_file_path = os.path.join(lib_path, lib_name)
+
+runtime_lib_dirs = []
+data_files = []
+
+if 'install' in sys.argv[1:]:
+    if sys.platform == "win32":
+        data_files = [('', [lib_file_path])]
+    else:
+        runtime_lib_dirs = [numpy.get_include(),] + AMGX_lib_dirs
+else:
+    # bdist_wheel or other subcommand
+    data_files = [(
+        os.path.relpath(get_python_lib(), sys.prefix), # should put the runtime under "lib\site-packages"
+        [lib_file_path]
```

Comment:
1. Is this essentially packaging the AMGX library into the wheel? I'm not sure about what issues this could cause on Windows, but on Linux, I'd be concerned about things like glibc binary compatibility or CUDA version compatibility. Perhaps, can we break this off into a separate PR and discuss there?

2. What about other `setup.py` commands like `python setup.py develop`?